### PR TITLE
[clipboard][web] Fix setStringAsync bouncing on Safari

### LINKED ISSUE #17337: Expo clipboard setString causes bouncing in ios Safari
### Summary

When using expo clipboard's setString method in safari on ios, the screen bounces up and then down as the text area is added and removed. 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Web

### SDK Version (managed workflow only)

44.0.6

### Environment

expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
      Yarn: 1.22.18 - ~/.nvm/versions/node/v16.14.2/bin/yarn
      npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    IDEs:
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.4 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      expo-cli: 5.3.0
    Expo Workflow: managed

### Reproducible demo

important note: the issue only appears when you access the snack in safari on ios--ie, go to the site and run the web preview in the safari browser on an iphone instead of in expo go. Press the Copy button and you should see the screen jump.

https://snack.expo.dev/@christopherleja/bossy-chocolates

### PR DESCRIPTION
**Pull request by barthap**
# Why

Closes #17337

# How

Did the thing described in https://github.com/expo/expo/issues/17337#issuecomment-1118210857 

It should not be breaking - falls back to current behavior if anything goes wrong

# Test Plan

Ran the snack from the issue in Safari on iOS. Also, the issue author confirmed that my fix worked

### CODE REVIEW

File: packages/expo-clipboard/src/web/ClipboardModule.ts [hunk 1]
```diff
@@ -103,8 +103,18 @@ export default {
...
+          if (!navigator.clipboard) {
+            throw new ClipboardUnavailableException();
+          }
+          await navigator.clipboard.writeText(text);
+          return true;
+        } catch {
+          // we can fall back to legacy behavior in any kind of failure
+          return this.setString(text);
+        }
+      }
```
Comment (bbarthec) [inline packages/expo-clipboard/src/web/ClipboardModule.ts:117]: I guess `ClipboardUnavailableException` should be propagated to the caller of this function right? 
```suggestion
        try {
          if (!navigator.clipboard) {
            throw new ClipboardUnavailableException(); // <- throwing that
          }
          await navigator.clipboard.writeText(text);
          return true;
        } catch { // <- would go here, is it intentional? 
          // we can fall back to legacy behavior in any kind of failure
          return this.setString(text);
        }
      }
```

**Review by bbarthec — COMMENTED** [review]
Looks almost ok. One question about `ClipboardUnavailableException` though 👀

File: packages/expo-clipboard/src/web/ClipboardModule.ts (hunk 1 above)
Comment (barthap) [reply packages/expo-clipboard/src/web/ClipboardModule.ts:117]: it is intentional - I can either throw or early return `this.setString`. In any kind of failure (including `navigator.clipboard` unavailability) I'd like to quietly use the legacy method. This way it is not breaking in any way.

Comment (bbarthec) [reply packages/expo-clipboard/src/web/ClipboardModule.ts:117]: Maybe add a little bit more of clarification to this catch statement then?

**Review by tsapeta — APPROVED** [review]

**Pull request merged**